src/providers/python.rs

```diff
@@ -0,0 +1,37 @@
+use anyhow::Result;
+
+use crate::{providers::Pkg, nixpacks::app::App};
+
+use super::Provider;
+
+pub struct PythonProvider {}
+impl Provider for PythonProvider {
+    fn name(&self) -> &str {
+        "python"
+    }
+
+    fn detect(&self, app: &crate::nixpacks::app::App) -> anyhow::Result<bool> {
+        Ok(app.includes_file("main.py"))
+    }
+
+    fn pkgs(&self, _app: &crate::nixpacks::app::App) -> Vec<super::Pkg> {
+        vec![
+            Pkg::new("pkgs.python38")
```

Comment:
This could check if the `runtime.txt` file exists, which contains the Python version to use. Example: https://github.com/heroku/heroku-buildpack-python/blob/main/spec/fixtures/pipenv_and_runtime_txt/runtime.txt

Reply:
Yeah, specific versioning of packages is going to be something quite hard to do.